test: Bluetooth: Mesh: subscription tree of extended models

alxelax opened the pull request:
PR adds testing of subscription of the tree of extended models.
Extended models should share subscription list among them.

Signed-off-by: Aleksandr Khromykh <Aleksandr.Khromykh@nordicsemi.no>

alxelax committed:
test: Bluetooth: Mesh: subscription tree of extended models

PR adds testing of subscription of the tree of extended models.
Extended models should share subscription list among them.

Signed-off-by: Aleksandr Khromykh <Aleksandr.Khromykh@nordicsemi.no>

trond-snekvik reviewed (commented):
I think there are some things that can be simplified here to make the test a bit easier to follow and maintain in the future. 

Also, it'd be good to have some coverage for corner cases in this. Don't need to expand the scope of this PR, but it could be good to keep in mind:
- Models across multiple elements (should not share subscription lists)
- Sibling models
- Subscription list entry allocation

alxelax commented:
> I think there are some things that can be simplified here to make the test a bit easier to follow and maintain in the future.
> 
> Also, it'd be good to have some coverage for corner cases in this. Don't need to expand the scope of this PR, but it could be good to keep in mind:
> 
> * Models across multiple elements (should not share subscription lists)
> * Sibling models
> * Subscription list entry allocation

I implemented a case that has been described in our backlog.
I added one more task for the corner cases in our internal backlog. Could you clarify a little bit more the last two points?
What did you mean:
Sibling models and Subscription list entry allocation
